app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginViewModel.kt

```diff
@@ -43,11 +47,24 @@ class LoginViewModel @Inject constructor(
     fun login(serverConfig: ServerConfig) {
         loginState = loginState.copy(loading = true, loginError = LoginError.None).updateLoginEnabled()
         viewModelScope.launch {
-            val loginResult = loginUseCase(loginState.userIdentifier.text, loginState.password.text, true, serverConfig)
+            val loginResult = loginUseCase(
+                userIdentifier = loginState.userIdentifier.text,
+                password = loginState.password.text,
+                shouldPersistClient = true,
+                serverConfig = serverConfig,
+                shouldStoreSession = false
+            )
             val loginError =
-                if (loginResult is AuthenticationResult.Success) registerClient(loginResult.userSession).toLoginError()
+                if (loginResult is AuthenticationResult.Success)
+                    registerClient(loginResult.userSession).let { registerClientResult ->
+                        if (registerClientResult is RegisterClientResult.Success) {
+                            saveSessionUseCase(loginResult.userSession)
```

Comment:
I think those two methods will be something we always will call together, so maybe it makes sense to wrap it in our own "use case" living inside the reloaded ?

Reply:
I reverted these changes because Mohamad is preparing AddAuthenticatedUserUseCase which will replace these two, so we will adapt it in the next PR. 😉